src/frontend/src/scheduler/distributed/query_manager.rs

```diff
@@ -106,8 +120,31 @@ impl QueryManager {
             }
         };
 
+        // TODO: Clean up queries status when ends. This should be done lazily.
+
         query_result_fetcher.collect_rows_from_channel(format).await
     }
+
+    pub fn cancel_queries_in_session(&self, session_id: SessionId) {
+        let mut write_guard = self.query_executions_map.lock().unwrap();
+        let values_iter = write_guard.values();
+        for query in values_iter {
+            let query = query.clone();
+            // spawn a task to abort. Avoid await point in this function.
+            tokio::spawn(async move { query.abort().await });
+        }
+        (*write_guard).retain(|_, query| query.session_id == session_id);
```

Comment:
Remove this?

Reply:
We can not. This is different from delete_query. When cancel a query, we should call abort. When a query ends, we should just delete it.

Reply:
1. The `spawn` will not block until the future finished. 
2. If you want to remove, you should call `remove` rather than `retain`.

Another thing is, I don't think we should obviously delete it here. It should be same with handling completion of a query.

Reply:
> The spawn will not block until the future finished.
Did not get this.

Yes, there's also metrics report in future, so we should not retain here. Fixed